docs/administrators/authorization.rst

```diff
@@ -0,0 +1,275 @@
+   * **noc**: Users with this role can do anything in a certain environment, except for modifying the desired state, changing environment settings and expert operations.
+   * **operator**: Users with this role can create, update and delete service instances in a certain environment next to all the actions allowed by the noc role.
+   * **environment-admin**: Users with this role can do anything in a certain environment, except for expert operations.
+   * **environment-expert-admin**: Users with this role can do anything in a certain environment.
+
+2. Global admin role: A user with this role can execute any API endpoint on the Inmanta server.
+
+The default policy makes the following assumptions about the content of the access token:
+
+* The ``sub`` claim is present and contains the username of the user the token is for.
+* Environment-scoped roles are defined in the ``urn:inmanta:roles`` claim of the access token. The value must be a dictionary that maps the ID of the environment to a list of roles the user has in that environment.
+* Global admins must have the claim ``urn:inmanta:is_admin`` in the access token with the value set to ``true``.
+
+
+Integration with database authentication
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The default policy integrates seamlessly with database authentication.
+
+* The admin user created using the ``/opt/inmanta/bin/inmanta-initial-user-setup`` command will have the ``urn:inmanta:is_admin`` claim set to true.
+* The web-console provides support to manage users and role assignments.
+
+If the policy contains a ``roles`` variable that contains a list role names, the Inmanta server will create these roles automatically when the server starts. The default policy defines this variable for the roles it can handle. Like that there is no need to create roles by hand. The snippet below provides an example. Removing a role from this list will not remove that role when the server starts. That should be done manually.
+
+
+.. code-block:: rego
+
+    package policy
+
+    # Define roles
+    roles := ["read-only", "noc", "operator", "environment-admin", "environment-expert-admin"]
+
+    # Get the metadata for the specific endpoint that is called.
+    endpoint_data := data.endpoints[input.request.endpoint_id]
+
+    # Don't allow anything that is not explicitly allowed.
+    default allow := false
+    ...
+
+
+Enable the policy-engine authorization provider
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+1. Set the :inmanta.config:option:`server.authorization-provider` config option to ``policy-engine``.
+2. Modify the default policy at ``/etc/inmanta/authorization/policy.rego`` if desired. More information about writing your own access policy is available :ref:`here<authorization-writing-own-policy>`.
+3. If a 3rd party auth broker is used, it must be configured to correctly set all claims required by the access policy.
```

Comment:
What is the consequence if these are not set? I assume the policy engine just fails? It wouldn't let things slip through, would it?

Reply:
If a certain claim is not set in the token, a lookup for that value in the policy will evaluate to `undefined`. Anything in the policy that depends on an undefined value will also evaluate to undefined. Undefined is a falsy value, so the allow rule will evaluate to False and reject the request.